### DISCUSSION TITLE
VPN Bind or Killswitch

### DISCUSSION THREAD

**Discussion by thob97** [haugene/docker-transmission-openvpn#2626] [Q&A]
I have a question regarding the setup this repo offers, and I would greatly appreciate any insights you can provide.

I was wondering if it directly binds transmission to the VPN, or if it incorporates a killswitch mechanism to disconnect the internet in case the VPN connection is lost. I've come across some information suggesting that killswitches may not be entirely secure, which has raised my curiosity about the specific approach employed in this setup.

Thank you kindly in advance for your assistance. I am grateful for any clarifications you can offer.

Best regards,
Thor

**Comment by edgd1er** [comment]
2 protections ( 1 optional) + 1 kill switch).
based on [route-up/route-pre-down](https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/#scripting-and-environmental-variables):
https://github.com/haugene/docker-transmission-openvpn/blob/42eb2ee94ef9a3ce45bdccb308f9387b36c4f6e0/openvpn/start.sh#L232

* tunnelUp script bind transmission to private ip:
https://github.com/haugene/docker-transmission-openvpn/blob/42eb2ee94ef9a3ce45bdccb308f9387b36c4f6e0/transmission/start.sh#L28-L32
* when openvpn disconnects, tunnelDown script is launched and transmission is killed (stop.sh):
 https://github.com/haugene/docker-transmission-openvpn/blob/42eb2ee94ef9a3ce45bdccb308f9387b36c4f6e0/transmission/stop.sh#L27-L34
* OPTIONAL: default route is removed after vpn is up. (off by default)
https://github.com/haugene/docker-transmission-openvpn/blob/42eb2ee94ef9a3ce45bdccb308f9387b36c4f6e0/transmission/start.sh#L90-L95

**Reply by thob97** [reply]
I want to express my sincere appreciation for your prompt response. Thank you very much.

If this script utilizes route-pre-down, then it should provide a secure solution, as it would halt transmission before removing the disconnected VPN connection. I'm curious why I've come across various sources stating that killswitches are considered unsafe. Perhaps other killswitches use different implementations which are less secure.

Furthermore, I would like to inquire whether you recommend utilizing both protections or if the default option alone is sufficient in ensuring a secure setup.

Once again, thank you for your assistance. I am grateful for any further insights you can provide.

Best regards

**Comment by edgd1er** [comment]
There are also questions when using wireguard (another vpn solution): https://unix.stackexchange.com/questions/733430/does-wireguard-postup-predown-really-work-as-a-kill-switch.
Killing the app on route change and binding it to the vpn interface is pretty safe for me. I don't see how on vpn termination, the application wound connect directly to the internet. bound interface is down, vpn ip is removed, and quickly transmission is killed. 

I also add a blocklist to transmission.

Deleting default route is an extra protection, depending on the risk of leaking you are willing to take.

You may run some tests with https://bash.ws/torrent-leak-test to check if leakage is detected when you disable the vpn interface ( ifconfig )
